pkg/configv2/types/compute.go

```diff
@@ -1,53 +1,40 @@
 package types
 
-// Compute represents the configuration for the compute service on the Bacalhau node.
-// It includes settings for enabling the service, connecting to orchestrators, TLS, heartbeat, store, capacity, and more.
 type Compute struct {
-	// Enabled when set to true will enable the compute service on the Bacalhau node.
-	Enabled bool `yaml:"Enabled,omitempty"`
-	// Orchestrators specifies a list of orchestrators the compute node will connect to.
+	Enabled       bool     `yaml:"Enabled,omitempty"`
 	Orchestrators []string `yaml:"Orchestrators,omitempty"`
-	// TLS specifies the TLS configuration used to connect to orchestrators.
-	TLS TLS `yaml:"TLS,omitempty"`
-	// Labels specifies a map of key value pairs the compute node will advertise to orchestrators.
-	Labels map[string]string `yaml:"Labels,omitempty"`
-	// Heartbeat specifies the compute node's heartbeat configuration.
-	Heartbeat Heartbeat `yaml:"Heartbeat,omitempty"`
-	// Capacity specifies the compute node's capacity configuration.
-	Capacity Capacity `yaml:"Capacity,omitempty"`
-	// Publishers specifies the configuration of publishers the compute node provides.
-	Publishers Publisher `yaml:"Publishers,omitempty"`
-	// Storages specifies the configuration of storages the compute node provides.
-	Storages Storage `yaml:"Storages,omitempty"`
-	// Engines specifies the configuration of engines the compute node provides.
-	Engines Engine `yaml:"Engines,omitempty"`
-	// Policy specifies the configuration of the compute node's job selection policy.
-	Policy SelectionPolicy `yaml:"Policy,omitempty"`
+	TLS           TLS      `yaml:"TLS,omitempty"`
+
+	// TODO what is this?
+	Authorization interface{} `yaml:"Authorization,omitempty"`
+
+	Heartbeat         Heartbeat         `yaml:"Heartbeat,omitempty"`
+	Labels            map[string]string `yaml:"Labels,omitempty"`
+	TotalCapacity     Resource          `yaml:"TotalCapacity,omitempty"`
+	AllocatedCapacity ResourceScaler    `yaml:"AllocatedCapacity,omitempty"`
```

Comment:
To clarify, Bacalhau calculates the available physical capacity and then allocate 80% of it for jobs. The original intention of the proposed configuration is to allow users to override the physical capacity instead of auto-calculating it, and then override the allocated %.

Meaning, a host with 5GB memory, but TotalCapacity of 1GB memory and AllocatedCapacity of 50%, will result in allocated 500MB for jobs.

This is a bit confusing, and that is why maybe just having a single AllocatedCapacity that can be either a % of static values will be much clearer. 

Meaning, a host with 5GB memory, and AllocatedCapacity of 50% will allocate 2.5GB, and AllocatedCapacity of 1GB will allocate 1GB